### ISSUE TITLE
Fails to start vertica on readonly filesystem

### ISSUE THREAD

**Issue by Jmoore1127** [vertica/vertica-kubernetes#267]
I am evaluating the operator in a dev environment. All of our environments use a readonly filesystem by default for security purposes. It seems the operator is having trouble starting the cluster. The pods in the stateful set have been created but I see repeated reconcile errors from the operator.

```
2022-10-14T18:20:03.917Z	INFO	controllers.VerticaDB	aborting reconcile of VerticaDB	{"verticadb": "data-infra-vertica/vertica-automessaging", "result": {"Requeue":false,"RequeueAfter":0}, "err": "open /tmp/accept_eula.py.2052615703: read-only file system"}

2022-10-14T18:20:03.917Z	ERROR	Reconciler error	{"controller": "verticaautoscaler", "controllerGroup": "vertica.com", "controllerKind": "VerticaAutoscaler", "verticaDB": {"name":"vertica-automessaging","namespace":"data-infra-vertica"}, "namespace": "data-infra-vertica", "name": "vertica-automessaging", "reconcileID": "a3dd0e7d-b26d-473a-935c-2e02d135438d", "error": "open /tmp/accept_eula.py.2052615703: read-only file system"}
```

Seems like this might be the issue? I can post a more complete log if needed.

**Comment by spilchen** [comment]
Are you deploying the operator with the official helm chart or via operatorhub.io? If so, this may be #257. That issue introduces a writable /tmp mount for cases where the root file system is read-only. But we haven't created a release yet with that fix. So, the only way to try that out is to run with an unreleased operator.

**Comment by Jmoore1127** [comment]
I'm deploying with the helm charts. I saw that issue. Was that resolving an issue with the operator not starting or the server pods not starting? I thought that was the operator itself failing to start.

**Comment by spilchen** [comment]
A customer had enabled `readOnlyRootFilesystem` in the security context and hit the same issue you saw. So, it is likely it will solve your issue too. Since you are only using it for dev purposes, I can give you instructions on how to deploy the operator that has this fix.

**Comment by Jmoore1127** [comment]
👍 that would be great. Also, any ETA on the next release that would include this?

**Comment by mspilly22** [comment]
> Also, any ETA on the next release that would include this?

We are targeting mid to late November.

Here are the instructions. I was hoping to give you a direct link to the artifacts, but GitHub doesn't allow that.

Download the release-artifacts from the this build of main:
https://github.com/vertica/vertica-kubernetes/actions/runs/3249491149

In a shell where the release-artifacts.zip was downloaded:
```
$ unzip release-artifacts.zip
$ kubectl apply -f verticaautoscalers.vertica.com-crd.yaml
$ kubectl apply -f verticadbs.vertica.com-crd.yaml
$ helm install vdb-op verticadb-operator-1.7.0.tgz --set image.repo=ghcr.io,image.name=vertica/verticadb-operator:5c7308333f32926234eaacbd0e73d59b4a6d807a
```

**Comment by spilchen** [comment]
> Also, any ETA on the next release that would include this?

We are targeting mid to late November.

Here are the instructions. I was hoping to give you a direct link to the artifacts, but GitHub doesn't allow that.

Download the release-artifacts from the this build of main:
https://github.com/vertica/vertica-kubernetes/actions/runs/3249491149

In a shell where the release-artifacts.zip was downloaded:
```
$ unzip release-artifacts.zip
$ kubectl apply -f verticaautoscalers.vertica.com-crd.yaml
$ kubectl apply -f verticadbs.vertica.com-crd.yaml
$ helm install vdb-op verticadb-operator-1.7.0.tgz --set image.repo=ghcr.io,image.name=vertica/verticadb-operator:5c7308333f32926234eaacbd0e73d59b4a6d807a
```

**Comment by Jmoore1127** [comment]
That has let it get farther. I'm facing another error that may or may not also be related to the readonly filesystem. 
```
2022-10-14T22:35:30.548Z	INFO	controllers.VerticaDB	ExecInPod stream	{"verticadb": "data-infra-vertica/vertica-automessaging", "pod": "data-infra-vertica/vertica-automessaging-default-0", "err": "command terminated with exit code 1", "stdout": "Attempting to retrieve file: [gs://my-bucket/metadata/VerticaDB/cluster_config.json]\n\nValidated 3-node database VerticaDB defined at communal storage gs://my-bucket/sms.\nCluster lease has expired.\n\nDatabase could not be revived.\nError: \nFailed to prepare a host for database participation.\nSee /opt/vertica/log/adminTools.log for details. Host hints follow.\n10.46.44.194 Permission Denied: '/vertica' \n10.46.13.9 Permission Denied: '/vertica' \n10.46.41.163 Permission Denied: '/vertica' \n\n", "stderr": ""}

2022-10-14T22:35:30.548Z	INFO	controllers.VerticaDB	Event logging	{"eventtype": "Warning", "reason": "ReviveDBPermissionDenied", "message": "revive_db failed because of a permission denied error.  Verify these paths match the ones used by the database: /data, /depot"}

**Comment by spilchen** [comment]
No, this isn't related. When you revive, the directories that were used when the db was created have to exist in the pod. You can control these with the `spec.local.dataPath` and `spec.local.depotPath` fields in the VerticaDB CR. What is the output of this query: `select node_name, storage_path, storage_usage from disk_storage;`

**Comment by Jmoore1127** [comment]
I don't have a running cluster at the moment. The marketplace solution has been spun down already. Is there a way I can pull that out of the cluster config file?

**Comment by spilchen** [comment]
The adminTools.log should have the info you are after. You can find this by dropping into the vertica pod and change to the  /opt/vertica/log directory. There should be an error message that includes the full path that was expected. Something like this:

```
dbadmin@vert-defaultsubcluster-0:/opt/vertica/log$ grep 'Permission denied.*target' adminTools.log  
2022-10-17 23:39:40.927 at_exec/257:0x7f95b2cd4740 [prepare_database._mkdir_impl] <ERROR> An exception was encountered during mkdir: [Errno 13] Permission denied: '/data', target was /data/vertdb/v_vertdb_node0001_catalog
```

You should be able to make a guess as to what the paths should be in the container. You can then set those in the VerticaDB with the `spec.local.dataPath` and `spec.local.depotPath` parms.

**Comment by spilchen** [comment]
I'm going to close this issue as the original issue seems to have been fixed with #257. We can continue the discussion about revive in the new issue you opened (#269)
